rir/src/ir/Compiler.cpp

```diff
@@ -458,497 +470,497 @@ bool compileSpecialCall(CompilerContext& ctx, SEXP ast, SEXP fun, SEXP args_, bo
         }
 
         // 2) Specialcalse normal assignment (ie. "i <- expr")
-        Match(lhs) {
-            Case(SYMSXP) {
-                cs << BC::guardNamePrimitive(fun);
-                compileExpr(ctx, rhs);
-                if (!voidContext) {
-                    // No ensureNamed needed, stvar already ensures named
-                    cs << BC::dup() << BC::invisible();
-                }
-                cs << (superAssign ? BC::stvarSuper(lhs) : BC::stvar(lhs));
-                return true;
-            }
-            Else(break)
-        }
-
-        // Find all parts of the lhs
-        SEXP target = nullptr;
-        l = lhs;
-        std::vector<SEXP> lhsParts;
-        while (!target) {
-            Match(l) {
-                Case(LANGSXP, fun, args) {
-                    assert(TYPEOF(fun) == SYMSXP);
-                    lhsParts.push_back(l);
-                    l = CAR(args);
-                }
-                Case(SYMSXP) {
-                    lhsParts.push_back(l);
-                    target = l;
-                }
-                Case(STRSXP) {
-                    assert(Rf_length(l) == 1);
-                    target = Rf_install(CHAR(STRING_ELT(l, 0)));
-                    lhsParts.push_back(target);
-                }
-                Else({
-                    errorcall(ast,
-                              "invalid (do_set) left-hand side to assignment");
-                })
-            }
+        Match(lhs){Case(SYMSXP){cs << BC::guardNamePrimitive(fun);
+        compileExpr(ctx, rhs);
+        if (!voidContext) {
+            // No ensureNamed needed, stvar already ensures named
+            cs << BC::dup() << BC::invisible();
         }
+        cs << (superAssign
+                   ? BC::stvarSuper(lhs)
+                   : BC::stvarCache(lhs, ctx.code.top()->cacheSlotFor(lhs)));
+        return true;
+    }
+    Else(break)
+}
 
-        // 3) Special case [ and [[
-
-        if (lhsParts.size() != 2) {
-            return false;
+// Find all parts of the lhs
```

Comment:
Whoa, what happened to the indentation here?

Reply:
hah, beatiful. yeah, I had to disable whitespaces in the diff settings to be able to review this PR at all. that's why I did not see it. I think this is the final argument why we should never run clang-format on the whole repo. @charig , can you please revert those whitespace changes?